webapp/channels/src/components/drafts/drafts_link/drafts_link.tsx

```diff
@@ -1,41 +1,82 @@
 import {FormattedMessage} from 'react-intl';
 import {useSelector, useDispatch} from 'react-redux';
 import {NavLink, useRouteMatch} from 'react-router-dom';
 
+import fetchTeamScheduledPosts from 'mattermost-redux/actions/scheduled_posts';
 import {syncedDraftsAreAllowedAndEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {
+    getScheduledPostsByTeamCount, hasScheduledPostError,
+} from 'mattermost-redux/selectors/entities/scheduled_posts';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 
 import {getDrafts} from 'actions/views/drafts';
 import {makeGetDraftsCount} from 'selectors/drafts';
 
+import {SCHEDULED_POST_URL_SUFFIX} from 'components/drafts/drafts';
 import DraftsTourTip from 'components/drafts/drafts_link/drafts_tour_tip/drafts_tour_tip';
 import ChannelMentionBadge from 'components/sidebar/sidebar_channel/channel_mention_badge';
 
+import type {GlobalState} from 'types/store';
+
 import './drafts_link.scss';
 
 const getDraftsCount = makeGetDraftsCount();
 
 function DraftsLink() {
     const dispatch = useDispatch();
 
+    const initialScheduledPostsLoaded = useRef(false);
+
     const syncedDraftsAllowedAndEnabled = useSelector(syncedDraftsAreAllowedAndEnabled);
-    const count = useSelector(getDraftsCount);
+    const draftCount = useSelector(getDraftsCount);
     const teamId = useSelector(getCurrentTeamId);
+    const teamScheduledPostCount = useSelector((state: GlobalState) => getScheduledPostsByTeamCount(state, teamId, true));
+
+    const itemsExist = draftCount > 0 || teamScheduledPostCount > 0;
+
+    const scheduledPostsHasError = useSelector((state: GlobalState) => hasScheduledPostError(state, teamId));
 
     const {url} = useRouteMatch();
     const isDraftUrlMatch = useRouteMatch('/:team/drafts');
+    const isScheduledPostUrlMatch = useRouteMatch('/:team/' + SCHEDULED_POST_URL_SUFFIX);
+
+    const urlMatches = isDraftUrlMatch || isScheduledPostUrlMatch;
 
     useEffect(() => {
         if (syncedDraftsAllowedAndEnabled) {
             dispatch(getDrafts(teamId));
         }
-    }, [teamId, syncedDraftsAllowedAndEnabled]);
+    }, [teamId, syncedDraftsAllowedAndEnabled, dispatch]);
 
-    if (!count && !isDraftUrlMatch) {
+    useEffect(() => {
+        const loadDMsAndGMs = !initialScheduledPostsLoaded.current;
+        dispatch(fetchTeamScheduledPosts(teamId, loadDMsAndGMs));
+        initialScheduledPostsLoaded.current = true;
+    }, [teamId]);
+
+    const pencilIcon = useMemo(() => {
+        return (
+            <i
+                data-testid='draftIcon'
+                className='icon icon-draft-indicator icon-pencil-outline'
+            />
+        );
+    }, []);
+
+    const scheduleIcon = useMemo(() => {
+        return (
+            <i
+                data-testid='scheduledPostIcon'
+                className='icon icon-draft-indicator icon-clock-send-outline'
+            />
+        );
+    }, []);
```

Comment:
Instead of `useMemo`, you can define these outside of the component. It takes a bit more of memory (since they will stay forever in memory) but reduce render time (doesn't have to call `useMemo`), reduce the memory footprint of the component (less hooks = less memory footprint), and simplifies the code.

Reply:
Done.